cetmix_tower_server/models/cx_tower_partner.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (C) 2025 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    server_ids = fields.One2many("cx.tower.server", "partner_id", string="Servers")
+    server_count = fields.Integer(
+        "Number of Servers", compute="_compute_server_count", store=True
+    )
+
+    secret_ids = fields.One2many(
+        "cx.tower.key.value",
+        "partner_id",
+        string="Secrets",
+    )
+
+    @api.depends("server_ids", "child_ids")
+    def _compute_server_count(self):
+        """Compute the total number of servers for a partner,
+        including all nested child partners, using child_of."""
+        Server = self.env["cx.tower.server"]
+        for partner in self:
```

Comment:
You already have all servers of the current record in the `server_ids`. So you can simply do [like this](https://github.com/cetmix/cetmix-tower/blob/964aa82fdb34f9e030a01db1fc22e183f84744e4/cetmix_tower_server/models/cx_tower_variable.py#L132).

If you wan to compute all the servers including the child records, you can simply sum the `server_count` values of all child records.

```python
for partner in self:
    server_count = len(partner.server_ids)
    if partner.child_ids:
        server_count += sum(child.server_count for child in partner.child_ids)
```

This is approach is better because it may benefit from using the cached records instead of doing a new query for each record every time.